File: Maximum-Number-of-Balloons.py
class Solution(object):
    def maxNumberOfBalloons(self, text):
        """
        :type text: str
        :rtype: int
        """
        b = a = l = o = n = 0
        
        for ch in text:
            if ch == 'b':
                b += 1
            elif ch == 'a':
                a += 1
            elif ch == 'l':
                l += 1
            elif ch == 'o':
                o += 1
            elif ch == 'n':
                n += 1
        
        return min(o // 2, l // 2, b, a, n)

File: Maximum-Number-of-Balloons.cpp
class Solution 
{
public:
    int maxNumberOfBalloons(string text) 
    {
        int b = 0, a = 0, l = 0, o = 0, n = 0;
        
        for (char ch : text) 
        {
            switch (ch) 
            {
                case 'b': b++; break;
                case 'a': a++; break;
                case 'l': l++; break;
                case 'o': o++; break;
                case 'n': n++; break;
            }
        }
        
        return min(min(o / 2, l / 2), min(min(b, a), n));
    }
};
